File: graphql/src/main/java/ru/romanow/protocols/graphql/GraphQLConfiguration.java
package ru.romanow.protocols.graphql;

import graphql.GraphQL;
import graphql.analysis.MaxQueryComplexityInstrumentation;
import graphql.analysis.MaxQueryDepthInstrumentation;
import graphql.execution.AsyncExecutionStrategy;
import graphql.execution.instrumentation.ChainedInstrumentation;
import graphql.schema.GraphQLSchema;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Arrays;

@Configuration
public class GraphQLConfiguration {

    @Bean
    @Autowired
    public GraphQL graphQL(GraphQLSchema graphQLSchema) {
        return GraphQL.newGraphQL(graphQLSchema)
                .queryExecutionStrategy(new AsyncExecutionStrategy())
                .instrumentation(new ChainedInstrumentation(Arrays.asList(
                        new MaxQueryComplexityInstrumentation(200),
                        new MaxQueryDepthInstrumentation(20)
                )))
                .build();
    }
}

File: graphql/src/main/java/ru/romanow/protocols/graphql/config/GraphQLConfiguration.kt
package ru.romanow.protocols.graphql.config

import graphql.schema.Coercing
import graphql.schema.GraphQLScalarType
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.graphql.execution.RuntimeWiringConfigurer


@Configuration
class GraphQLConfiguration {

    @Bean
    fun runtimeWiringConfigurer(): RuntimeWiringConfigurer {
        return RuntimeWiringConfigurer { it.scalar(voidScalar()) }
    }

    @Bean
    fun voidScalar(): GraphQLScalarType {
        return GraphQLScalarType.newScalar()
            .name("Void")
            .description("Empty result")
            .coercing(object : Coercing<Unit, Unit> {})
            .build()
    }
}
